blobfuse/src/DataLakeBfsClient.cpp

```diff
@@ -325,24 +325,27 @@ std::vector<std::string> DataLakeBfsClient::Rename(const std::string /*sourcePat
 
 std::vector<std::string> DataLakeBfsClient::Rename(std::string sourcePath, std::string destinationPath)
 {
+    std::vector<std::string> file_paths_to_remove;
+
     errno = 0;
     m_adls_client->move_file(
             configurations.containerName,
             sourcePath.substr(1),
             configurations.containerName,
             destinationPath.substr(1));
-
-    std::vector<std::string> file_paths_to_remove;
-    std::string srcMntPathString = prepend_mnt_path_string(sourcePath);
-    std::string dstMntPathString = prepend_mnt_path_string(destinationPath);
-
-    long int rename_ret = rename_cached_file(srcMntPathString.c_str(), dstMntPathString.c_str());
-
-    if(rename_ret != 0)
-    {
-        syslog(LOG_ERR, "Failure to rename source file %s in the local cache.  Errno = %d.\n", sourcePath.c_str(), errno);
+    
+    if(errno != 0) {
+        syslog(LOG_ERR, "Failure to rename source file %s in container.  Errno = %d.\n", sourcePath.c_str(), errno);
+    } else {
+        std::string srcMntPathString = prepend_mnt_path_string(sourcePath);
+        std::string dstMntPathString = prepend_mnt_path_string(destinationPath);
+
+        long int rename_ret = rename_cached_file(srcMntPathString.c_str(), dstMntPathString.c_str());
+        if(rename_ret != 0) {
+            syslog(LOG_ERR, "Failure to rename source file %s in the local cache.  Errno = %d.\n", sourcePath.c_str(), errno);
+        } 
+        file_paths_to_remove.push_back(sourcePath);
```

Comment:
sourcePath should not be added to the file_paths_to_remove if the file does not exist in cache. So the return from rename_cached_file should return a tuple, the second value of the pair should indicate if the file was in cache or not.

Reply:
Added a check to push to vector only when local cache was cleared.